feat(third-parties): add consent management support to GoogleTagManager

NimaC opened the pull request:
## What?

Adds consent management support to GoogleTagManager component by making it more flexible to support different CMP integration approaches.

## Why?

**Component Inflexibility**: The current GoogleTagManager component only supports standard script execution, but some CMPs require additional script attributes for integration. This forces developers to abandon the Next.js component for manual script injection.

**CMP Integration Requirements**: Some CMPs require:
- `type="text/plain"` to prevent execution until consent
- Platform-specific `data-*` attributes for identification

**Known Compatible CMPs:**
- ✅ **Cookiebot**: `data-cookieconsent="statistics"`
- ✅ **Usercentrics**: `data-usercentrics="Google Tag Manager"`
- ✅ **Others**: Any CMP using similar HTML attribute patterns

## How?

**API (Backward Compatible):**
```jsx
// Standard GTM (unchanged)
<GoogleTagManager gtmId="GTM-XYZ" />

// With CMP integration
<GoogleTagManager
  gtmId="GTM-XYZ"
  type="text/plain"
  data-usercentrics="Google Tag Manager"
/>
```

**Implementation:**
- Added `type` prop for script execution control
- Added support for `data-*` attributes for CMP integration
- Both initialization and external scripts receive same attributes
- Zero breaking changes

## Benefits

- **Enables CMP Integration**: Supports CMPs that require script-level attributes
- **Component Flexibility**: Developers can use Next.js component with their chosen CMP
- **Future-Proof**: Works with any CMP using standard HTML attributes
- **Developer Experience**: Clean API following React patterns

NimaC committed:
feat(third-parties): add consent management support to GoogleTagManager

- Add type prop to control script execution (application/javascript | text/plain)
- Add support for data-* attributes for CMP platform integration
- Support all major CMP platforms (Usercentrics, OneTrust, Cookiebot, Didomi)
- Add comprehensive E2E tests for consent management scenarios
- Update documentation with consent management examples
- Maintain full backward compatibility

Closes #XXXXX

ijjk commented:
Allow CI Workflow Run

- [ ] approve CI run for commit: c121cc69e00b96c159c757a81850073740f4a99a

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

NimaC committed:
refactor(tests): improve consent management test with data-driven approach

- Replace verbose individual tests with concise loop-based testing
- Use structured test data for better maintainability
- Reduce code duplication while maintaining comprehensive coverage
- Improve readability and make it easier to add new CMP scenarios

NimaC committed:
improve test

NimaC committed:
refactor(tests): simplify consent management test to single representative case

- Reduce test complexity by using one CMP platform (Usercentrics) as representative
- Remove redundant test cases since all CMPs follow the same pattern
- Focus on testing core functionality: type prop and data-* attributes
- Improve test maintainability and execution speed
- Maintain comprehensive coverage of the feature being tested

NimaC committed:
refactor(tests): separate consent management test from standard GTM test

- Remove redundant standard GTM test from gtm-consent test case
- Focus gtm-consent test only on new consent management functionality
- Maintain proper separation of concerns between gtm.js and gtm-consent.js
- Standard GTM behavior is already covered by existing gtm test
- Consent management test now only validates type=text/plain and data-* attributes

NimaC committed:
docs(readme): simplify consent management examples to single representative case

- Replace multiple verbose code examples with one clear Usercentrics example
- List other CMP platforms with their data attributes in concise format
- Reduce documentation length while maintaining comprehensive coverage
- Focus on the pattern rather than repetitive code blocks
- Improve readability and maintainability of documentation

NimaC commented:
Hi @ijjk! Thank you for reviewing this PR. 

This PR is ready for CI approval. It adds consent management support to GoogleTagManager, addressing community requests from discussions #64497 and #53868.

Key highlights:
- ✅ Zero breaking changes - fully backward compatible
- ✅ Comprehensive E2E tests included  
- ✅ Universal CMP compatibility (Usercentrics, OneTrust, Cookiebot, etc.)
- ✅ Addresses frequently requested GDPR/CCPA compliance feature

The implementation is conservative and focused, using standard HTML attributes that CMPs already expect. Ready for CI and review whenever convenient.

Thanks for maintaining this amazing project! 🙏

NimaC commented:
Hey @ijjk, thanks for reviewing this PR, its ready for a CI run now

NimaC committed:
Merge branch 'canary' into feat/gtm-consent-management

NimaC committed:
Merge branch 'canary' into feat/gtm-consent-management
